pkg/controller/servicecidrs/servicecidrs_controller.go

```diff
@@ -464,6 +460,41 @@ func (c *Controller) removeServiceCIDRFinalizerIfNeeded(ctx context.Context, cid
 	return nil
 }
 
+// updateConditionIfNeeded updates the status condition of the ServiceCIDR if needed.
+func (c *Controller) updateConditionIfNeeded(ctx context.Context, cidr *networkingapiv1.ServiceCIDR, newCondition metav1.Condition) error {
```

Comment:
seems like the generic helper methods FindStatusCondition / SetStatusCondition would let you avoid duplicating some of this logic